src/main/java/com/launchdarkly/client/StreamProcessor.java

```diff
@@ -171,4 +195,21 @@ int version() {
     }
 
   }
+
+  private final class HeartbeatDetector implements Runnable {
+
+    @Override
+    public void run() {
+      DateTime fiveMinutesAgo = DateTime.now().minusSeconds(DEAD_CONNECTION_INTERVAL_SECONDS);
+      if (lastHeartbeat.isBefore(fiveMinutesAgo) && es.getState() == ReadyState.OPEN) {
+        try {
+          logger.info("Stream stopped receiving heartbeats- reconnecting.");
+          es.close();
+          start();
```

Comment:
Looking at the code, it's not really possible. Nothing in `es.close()` throws an exception-- it's simply there to implement the `Closeable` interface. Nevertheless, calling `start` in `finally` makes sense in case the exception semantics in `close` change.